readthedocs/projects/tasks.py

```diff
@@ -368,11 +368,13 @@ def update_app_instances(self, html=False, localmedia=False, search=False,
                 )
             )
 
-            # Delayed tasks
-            # TODO these should be chained on to the broadcast calls. The
-            # broadcast calls could be lumped together into a promise, and on
-            # task result, these next few tasks can be updated, also in a
-            # chained fashion
+            # We're waiting to make sure all the file syncing has happened,
+            # so we can perform other options on those files without a race conflict.
+            # This has a low (5 second * 4) timeout, so we don't wait forever,
+            # but should capture most failure cases for now.
+            for result in task_results:
+                task.get(timeout=5)
```

Comment:
I assume `task.get` -> `result.get` here. But with `broadcast` returning a celery `GroupReult` this can also be:

```
job = broadcast(...)
job.get(timeout=5)
```

Notes on using `chord()` and removing locking from application side probably makes most sense.

If this doesn't address pushing task execution onto chained tasks, the removed comment above also probably still stands.